src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,186 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    public static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> previous;
+
+        public Node(Node<T> prev, T item, Node<T> next) {
+            this.previous = prev;
+            this.item = item;
+            this.next = next;
+        }
+    }
+
+    private Node<T> firstNode;
+    private Node<T> lastNode;
+    private int size;
+
+    public MyLinkedList() {
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        if (isPositionIndex(index)) {
+            if (index == size) {
+                linkLast(value);
+            } else {
+                linkBefore(value, node(index));
+            }
+        } else {
+            throw new IndexOutOfBoundsException(" ");
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list == null || list.size() == 0) {
+            throw new NullPointerException(" ");
+        } else {
+            for (T t : list) {
+                add(t);
+            }
+        }
+
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (isElementIndex(index)) {
+            return node(index).item;
+        } else {
+            throw new IndexOutOfBoundsException(" ");
+        }
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        if (isElementIndex(index)) {
+            Node<T> node = node(index);
+            T old = node.item;
+            node.item = value;
+            return old;
+        } else {
+            throw new IndexOutOfBoundsException(" ");
+        }
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (isElementIndex(index)) {
+            return unlink(node(index));
+        } else {
+            throw new IndexOutOfBoundsException(" ");
+        }
     }
 
     @Override
     public boolean remove(T object) {
+        if (object != null) {
+            for (Node<T> x = firstNode; x != null; x = x.next) {
+                if (object.equals(x.item)) {
+                    unlink(x);
+                    return true;
+                }
+            }
+        } else {
+            for (Node<T> x = firstNode; x != null; x = x.next) {
+                if (x.item == null) {
+                    unlink(x);
+                    return true;
+                }
+            }
+        }
         return false;
     }
```

Comment:
x isn't the best variable name
Also, we don't need to change any of the fields in the node we are unlinking (x in this case, I believe)

We have almost identical for blocks, can they be merged?